subprojects/smoke-test/src/smokeTest/groovy/org/gradle/smoketests/AbstractSmokeTest.groovy

```diff
@@ -262,6 +263,9 @@ abstract class AbstractSmokeTest extends Specification {
             .withTestKitDir(IntegrationTestBuildContext.INSTANCE.gradleUserHomeDir)
             .withProjectDir(testProjectDir)
             .forwardOutput()
+            .withArguments("-Porg.gradle.java.installations.paths=${AvailableJavaHomes.getAvailableJvms().collect { it.javaHome.absolutePath }.join(",")}")
+            .withArguments("-Porg.gradle.java.installations.auto-detect=false")
+            .withArguments("-Porg.gradle.java.installations.auto-download=false")
```

Comment:
The test is still flaky: https://ge.gradle.org/s/glphel6qrwlfe/tests/:smoke-test:gradleBuildSmokeTest/org.gradle.smoketests.GradleBuildSmokeTest/can%20build%20Gradle%20distribution?top-execution=1

I think the problem is that `withArguments` sets all arguments, so you'd need to add this list similarly to the other arguments below.

Reply:
Made the change.